configurate-core/src/main/java/ninja/leaping/configurate/objectmapping/serialize/TypeSerializerCollection.java

```diff
@@ -97,12 +98,50 @@ private RegisteredSerializer(Predicate<TypeToken<?>> predicate, TypeSerializer<?
         }
 
         private RegisteredSerializer(TypeToken<?> type, TypeSerializer<?> serializer) {
-            this(type::isSupertypeOf, serializer);
+            this(new SuperTypePredicate(type), serializer);
         }
     }
 
-    private final class SerializerList extends CopyOnWriteArrayList<RegisteredSerializer>
-            implements Function<TypeToken<?>, TypeSerializer<?>> {
+    /**
+     * Effectively a predicate which is <code>type::isSupertypeOf</code>.
+     *
+     * <p>The isSupertypeOf method was only added in Guava 19.0, and was previously named
+     * isAssignableFrom.</p>
+     */
+    private static final class SuperTypePredicate implements Predicate<TypeToken<?>> {
+        private static final Method SUPERTYPE_TEST;
+        static {
+            Method supertypeTest;
+            try {
+                supertypeTest = TypeToken.class.getMethod("isSupertypeOf", TypeToken.class);
+            } catch (NoSuchMethodException e) {
+                try {
+                    supertypeTest = TypeToken.class.getMethod("isAssignableFrom", TypeToken.class);
+                } catch (NoSuchMethodException ignored) {
+                    supertypeTest = null;
+                }
+            }
+            SUPERTYPE_TEST = Objects.requireNonNull(supertypeTest);
```

Comment:
pass a message here, just in case this is ever reached

Reply:
Changed